### PR TITLE
lib/sdt_task.c: separate generic allocation code from task_ctx allocation

### PR DESCRIPTION
**Pull request by etsal**
The arena allocator currently allocates a single element per task because it addresses the need for dynamic task_ctx allocation. However, arena allocation is useful for other data structures that are not 1:1 correlated to tasks, e.g., domains, bitmasks. Separate the allocator into two pieces: A general allocation API backed by arena memory, and a task_ctx-specific API.

### CODE REVIEW

File: lib/sdt_task.bpf.c [hunk 1]
```diff
@@ -391,7 +391,6 @@ static SDT_TASK_FN_ATTRS void sdt_task_free_idx(__u64 idx)
 
 		data[pos] = (struct sdt_task_data) {
 			.tid.gen = data->tid.gen + 1,
-			.tptr = 0,
```
Comment (htejun) [inline lib/sdt_task.bpf.c:394]: Why is `.tptr` being removed?

**Review by htejun — APPROVED** [review]

File: lib/sdt_task.bpf.c (hunk 1 above)
Comment (etsal) [reply lib/sdt_task.bpf.c:394]: We are currently not using it. I can instead properly populate it to make debugging easier for the task allocator (AFAICT that was its initial purpose, right)?

Comment (htejun) [reply lib/sdt_task.bpf.c:394]: IIRC, the intention was enabling easily looking up task struct from tid. How would you go from tid to task without the field?

Comment (etsal) [reply lib/sdt_task.bpf.c:394]: Ah I see, yeah the caller of the API passes ```struct task_struct *```  so they would already have the tptr available, but other readers of the struct like userspace wouldn't have a way of finding which task the data belongs to if reading it straight from the allocator.

**Review by htejun — APPROVED** [review]

**Comment by hodgesds** [comment]
I think I see some compilation issues with cargo:
```
error: failed to run custom build command for `scx_rusty v1.0.8 (/home/daniel/git/scx/scheds/rust/scx_rusty)`

Caused by:
  process didn't exit successfully: `/home/daniel/git/scx/target/release/build/scx_rusty-0d72b9db39b211ea/build-script-build` (exit status: 101)
  --- stdout
  scx_utils:clang=ClangInfo { clang: "clang", ver: "19.1.4", arch: "x86_64" } ["-g", "-O2", "-Wall", "-Wno-compare-distinct-pointer-types", "-D__TARGET_ARCH_x86", "-mcpu=v3", "-mlittle-endian", "-idirafter", "/usr/lib/llvm/19/bin/../../../../lib/clang/19/include", "-idirafter", "/usr/local/include", "-idirafter", "/usr/include", "-I/home/daniel/git/scx/target/release/build/scx_rusty-702311ab341434af/out/scx_utils-bpf_h/arch/x86", "-I/home/daniel/git/scx/target/release/build/scx_rusty-702311ab341434af/out/scx_utils-bpf_h", "-I/home/daniel/git/scx/target/release/build/scx_rusty-702311ab341434af/out/scx_utils-bpf_h/bpf-compat"]
  clang is version 19.1.4
  Building ../../../lib/sdt_alloc.bpf.c

  --- stderr
  thread 'main' panicked at scheds/rust/scx_rusty/build.rs:14:10:
  called `Result::unwrap()` on an `Err` value: failed to build `../../../lib/sdt_alloc.bpf.c`

  Caused by:
      0: Failed to compile /home/daniel/git/scx/target/release/build/scx_rusty-702311ab341434af/out/bpf from ../../../lib/sdt_alloc.bpf.c
      1: Command `clang -g -O2 -Wall -Wno-compare-distinct-pointer-types -D__TARGET_ARCH_x86 -mcpu=v3 -mlittle-endian -idirafter /usr/lib/llvm/19/bin/../../../../lib/clang/19/include -idirafter /usr/local/include -idirafter /usr/include -I/home/daniel/git/scx/target/release/build/scx_rusty-702311ab341434af/out/scx_utils-bpf_h/arch/x86 -I/home/daniel/git/scx/target/release/build/scx_rusty-702311ab341434af/out/scx_utils-bpf_h -I/home/daniel/git/scx/target/release/build/scx_rusty-702311ab341434af/out/scx_utils-bpf_h/bpf-compat -I /tmp/.tmpMmNGrW/bpf/src -fno-stack-protector -g -O2 -target bpf -c ../../../lib/sdt_alloc.bpf.c -o /home/daniel/git/scx/target/release/build/scx_rusty-702311ab341434af/out/bpf` failed (exit status: 1)
      2: In file included from ../../../lib/sdt_alloc.bpf.c:9:
         /home/daniel/git/scx/target/release/build/scx_rusty-702311ab341434af/out/scx_utils-bpf_h/lib/sdt_task.h:109:25: warning: multiple identical address spaces specified for type [-Wduplicate-decl-specifier]
           109 | struct sdt_data __arena __arena *sdt_alloc(struct sdt_allocator *alloc);
               |                         ^
         /home/daniel/git/scx/target/release/build/scx_rusty-702311ab341434af/out/scx_utils-bpf_h/scx/bpf_arena_common.h:36:32: note: expanded from macro '__arena'
            36 | #define __arena __attribute__((address_space(1)))
               |                                ^
         ../../../lib/sdt_alloc.bpf.c:491:49: error: automatic variable qualified with an address space
           491 |         struct sdt_data __arena *data = NULL, __arena *val;
               |                                                        ^
         1 warning and 1 error generated.

  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

**Comment by etsal** [comment]
Ah I think this is due to Clang 19, I'll fix it up

**Pull request merged**